tests/job/test_genericJob.py

```diff
@@ -502,5 +504,56 @@ def test_return_codes(self):
             pass
         self.assertTrue(j.status.aborted, "Job did not abort even though return code is 2!")
 
+    def test_job_executor_set(self):
+        j = self.project.create_job(ReturnCodeJob, "job_with_executor_set")
+        j.server.executor = ProcessPoolExecutor()
+        self.assertTrue(j.server.run_mode.executor)
+        j.server.run_mode.modal = True
+        self.assertFalse(j.server.run_mode.executor)
```

Comment:
```suggestion
        self.assertFalse(j.server.run_mode.executor)
        self.assertIsNone(j.server.executor)
```

Reply:
This part should also include tests to verify that whatever behaviour we want for `Server.future` when the run mode is changed away from `executor` are actually happening

Reply:
Again, changing the `run_mode` during or after the execution, when a future object exists sounds very dangerous to me.

Reply:
> Again, changing the `run_mode` during or after the execution, when a future object exists sounds very dangerous to me.

I totally agree, I wouldn't do it. But we let users do it. My point is that I would like a 1:1 relationship between `server.executor is not None` and `server.run_mode.executor` under all operations that the user can permissibly execute. The behaviour in `Server.executor` setter and getter do this now, so I just want to test that it stays that way.